README.md

```diff
@@ -120,7 +120,123 @@ The transformer will also generate a file named `utils.js` inside the wrapper cl
+
+Implementation class files contain a single export, `implementation`, whose value is the implementation class. (See [#59](https://github.com/jsdom/webidl2js/issues/59) for potentially making this the default export instead.) The class will contain the following elements:
+
+### The constructor
+
+A constructor for your implementation class, with signature `(constructorArgs, privateData)` can serve several purposes:
+
+- Setting up initial state that will always be used, such as caches or default values
+- Processing constructor arguments `constructorArgs` passed to the wrapper class constructor, if the interface in question has a `[Constructor]` extended attribute.
+- Processing any private data `privateData` which is provided when other parts of your program use the generated `create()` or `createImpl()` exports of the wrapper class file. This is useful for constructing instances with specific state that cannot be constructed via the wrapper class constructor.
+
+If you don't need to do any of these things, the constructor is entirely optional.
+
+Additionally, the constructor should not be provided for mixin classes; it will never be called.
+
+### Methods implementing IDL operations
+
+IDL operations that you wish to implement need to have corresponding methods on the implementation class.
+
+The wrapper class will take care of type conversions for the arguments, so you can operate on the incoming data assuming it is converted to the appropriate Web IDL type. You can also assume that your `this` value is an implementation class; appropriate brand-checking was performed by the wrapper class.
+
+The wrapper class will even convert any implementation classes you return to the corresponding wrapper classes. This allows you to usually stay entirely within the realm of impls and never deal with wrappers yourself.
+
+However, note that apart from Web IDL container return values, this impl-back-to-wrapper conversion doesn't work in a "deep" fashion. That is, if you directly return an impl, or return an array or object containing impls from a `sequence<>`, `FrozenArray<>`, or `record<>`-returning operation, the conversion will work. But if you return some other container (such as your own `NodeList` implementation) containing impls, the impl classes will escape to your consumers. Caution is advised.
+
+One other subtlety here is overloads: if the IDL file defines overloads for a given operation, webidl2js currently does not dispatch to separate implementation class methods, but instead performs the overload resolution algorithm and then sends the result to the same backing method. We're considering changing this ([#29](https://github.com/jsdom/webidl2js/issues/29)), but in the meantime, properly implementing overloads requires doing some extra type-checking (often using appropriate `isImpl()` functions) to determine which case of the overload you ended up in.
```

Comment:
A word of caution. For i

```webidl
void op(byte a);
void op(URL a);
```

The overload resolution in webidl2js currently just throws up its hands and say "oh well let's leave conversion to the impl class". In that case manual unwrapping is needed.